File: data-types-type-system/DataTypesTypeSystem.cpp
#include <iostream>
#include <map>
#include <optional>
#include <string>
#include <vector>

int main() {
    int age = 28;
    double temperature = 36.7;
    std::string name = "Nazmul";
    bool is_active = true;
    std::vector<std::string> tags{"cpp", "types", "demo"};
    std::map<std::string, int> scores{{"math", 90}, {"science", 88}};
    std::optional<std::string> middle_name = std::nullopt;

    std::string converted_age = std::to_string(age);
    int parsed_number = std::stoi("42");

    std::cout << "Language: C++\n";
    std::cout << "age=" << age << " (int)\n";
    std::cout << "temperature=" << temperature << " (double)\n";
    std::cout << "name=" << name << " (std::string)\n";
    std::cout << "is_active=" << is_active << " (bool)\n";

    std::cout << "tags=[";
    for (std::size_t i = 0; i < tags.size(); ++i) {
        std::cout << tags[i] << (i + 1 < tags.size() ? ", " : "");
    }
    std::cout << "] (std::vector<std::string>)\n";

    std::cout << "scores=math:" << scores["math"] << ", science:" << scores["science"]
              << " (std::map<std::string, int>)\n";
    std::cout << "middle_name=" << (middle_name.has_value() ? middle_name.value() : "null")
              << " (std::optional<std::string>)\n";
    std::cout << "converted_age=" << converted_age << " (std::string)\n";
    std::cout << "parsed_number=" << parsed_number << " (int)\n";

    return 0;
}

File: data-types-type-system/DataTypesTypeSystem.cs
using System;
using System.Collections.Generic;

class DataTypesTypeSystem
{
    static void Main()
    {
        int age = 28;
        double temperature = 36.7;
        string name = "Nazmul";
        bool isActive = true;
        var tags = new List<string> { "csharp", "types", "demo" };
        var scores = new Dictionary<string, int> { { "math", 90 }, { "science", 88 } };
        string? middleName = null;

        string convertedAge = age.ToString();
        int parsedNumber = int.Parse("42");

        Console.WriteLine("Language: C#");
        Console.WriteLine($"age={age} (int)");
        Console.WriteLine($"temperature={temperature} (double)");
        Console.WriteLine($"name={name} (string)");
        Console.WriteLine($"isActive={isActive} (bool)");
        Console.WriteLine($"tags=[{string.Join(", ", tags)}] (List<string>)");
        Console.WriteLine($"scores=math:{scores["math"]}, science:{scores["science"]} (Dictionary<string, int>)");
        Console.WriteLine($"middleName={middleName} (string?)");
        Console.WriteLine($"convertedAge={convertedAge} (string)");
        Console.WriteLine($"parsedNumber={parsedNumber} (int)");
    }
}
